magento/magento2
# Add VIES validation result object to the response.

### Description
This commit adds `result_object` to the method response. This allows doing some additional validation 
logic in the plugin written for `Magento\Customer\Model\Vat` model.
For some countries, this object returns additional fields:

```
traderName: "customer or company name"
traderCompanyType: "company type"
traderAddress: "
multiline address
associated with tax number
"
```

### Manual testing scenarios
1. Apply the patch.
2. To test it you need a valid Netherland taxvat number.
3. Open vat validator model and add the following code right before the end of the method: https://github.com/magento/magento2/blob/HEAD/app/code/Magento/Customer/Model/Vat.php#L218

    ```php
    (new \Zend\Log\Logger())
        ->addWriter(new \Zend\Log\Writer\Stream(BP . '/var/log/custom.log'))
        ->debug((array)$gatewayResponse->getResultObject());
    ```

4. Go to admin config _Stores > Configuration > General > Store Information_
5. Select a valid country for your tax number, fill the tax number field, and press the "Validate VAT Number" button.
6. Check the `magento_root/var/log/custom.log`. You should see the VIES result object:

    ```php
    2020-11-23T09:25:57+00:00 DEBUG (7): array (
      'countryCode' => 'NL',
      'vatNumber' => 'vatnumber',
      'requestDate' => '2020-11-23+01:00',
      'valid' => true,
      'traderName' => 'companyname',
      'traderCompanyType' => '---',
      'traderAddress' => '
    company
    address
    ',
      'requestIdentifier' => '',
    )

    ```

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#38511: Add VIES validation result object to the response.
